src/ComputeSharp.D2D1.SourceGenerators/SyntaxRewriters/HlslSourceRewriter.cs

```diff
@@ -202,9 +202,22 @@ public override SyntaxNode VisitDefaultExpression(DefaultExpressionSyntax node)
 
                 return updatedNode.WithToken(Literal(literal, 0f));
             }
-            else if (type.GetFullMetadataName().Equals(typeof(double).FullName))
+            else if (type.SpecialType == SpecialType.System_Double)
             {
-                return updatedNode.WithToken(Literal(updatedNode.Token.ValueText + "L", 0d));
+                string literal = updatedNode.Token.ValueText;
+
+                // If the numeric literal is neither a decimal nor an exponential, add the ".0L" suffix.
+                // This is necessary because otherwise integer literals would actually be of type long.
+                if (literal.IndexOfAny(new[] { '.', 'E' }) == -1)
```

Comment:
isn't lowercase `e` also permitted?

Also won't this allocate an array on every inspected literal? Should that array be cached in a `static readonly`?

Reply:
Used an array in https://github.com/Sergio0694/ComputeSharp/pull/301/commits/f66342d21256b7c5a24164198a25a7705101579d and added some unit tests for `e` literals (no need to check for the character as that text is produced by Roslyn, and it always uses the uppercase `E` when creating the string, as it formats it then).